executor/batchsubmitter/batch.go

```diff
@@ -267,3 +269,49 @@ func prependLength(data []byte) []byte {
 	binary.LittleEndian.PutUint64(lengthBytes, uint64(len(data)))
 	return append(lengthBytes, data...)
 }
+
+func (bs *BatchSubmitter) submitGenesis(ctx types.Context) error {
+	res, err := bs.node.GetRPCClient().Genesis(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to query genesis")
+	}
+
+	genesisBz, err := json.Marshal(res.Genesis)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal genesis")
+	}
+
+	chunkLength := int(math.Ceil(float64(len(genesisBz)) / float64(bs.batchCfg.MaxChunkSize)))
```

Comment:
```
// Calculate chunk length using integer ceiling division
chunkLength := (totalSize + int(bs.batchCfg.MaxChunkSize) - 1) / int(bs.batchCfg.MaxChunkSize)
```

This change avoids unnecessary type conversions between int and float64, making the calculation more straightforward and efficient while achieving the same result. Let me know what you think!